File: src/main/java/tokens/InvalidDataException.java
package tokens;

@SuppressWarnings("serial")
public class InvalidDataException extends RuntimeException {
	public InvalidDataException() {
		super();
	}

	public InvalidDataException(String message, Throwable cause) {
		super(message, cause);
	}

	public InvalidDataException(String message) {
		super(message);
	}

	public InvalidDataException(Throwable cause) {
		super(cause);
	}
}

File: src/main/scala/serialization/InvalidDataException.scala
package serialization

import auth.AuthenticationException

class InvalidDataException(message: String, cause: Throwable) extends AuthenticationException(message, cause) {
  def this(message: String) = this(message, null)
  def this(cause: Throwable) = this(null, cause)
  def this() = this(null, null)
}
